py-rattler-build/src/lib.rs

```diff
@@ -37,6 +45,174 @@ where
     })
 }
 
+/// Python wrapper for SelectorConfig
+#[pyclass]
+#[derive(Clone)]
+pub struct PySelectorConfig {
+    inner: SelectorConfig,
+}
+
+#[pymethods]
+impl PySelectorConfig {
+    #[new]
+    #[pyo3(signature = (target_platform=None, host_platform=None, build_platform=None, variant=None, experimental=None, allow_undefined=None, recipe_path=None))]
+    fn new(
+        target_platform: Option<String>,
+        host_platform: Option<String>,
+        build_platform: Option<String>,
+        variant: Option<HashMap<String, Py<PyAny>>>,
+        experimental: Option<bool>,
+        allow_undefined: Option<bool>,
+        recipe_path: Option<PathBuf>,
+    ) -> PyResult<Self> {
+        let target_platform = target_platform
+            .map(|p| Platform::from_str(&p))
+            .transpose()
+            .map_err(|e| PyRuntimeError::new_err(e.to_string()))?
```

Comment:
I think it is better to pack them in a more concrete error, for example, PyRattlerBuildError and make this into exception rather than a RuntimeError.

You can take a look how we have done it in py-rattler:
https://github.com/conda/rattler/blob/main/py-rattler/src/error.rs#L102